AbstractReadWriteTest throws incorrect sequence error after segment rebalance between readers.

**Problem description**
AbstractReadWriteTest  throws an incorrect `Event order violation` when segments are rebalanced between readers.

```
// reader 1 reads event 10.
2019-01-21 09:20:30,363 364060 [ReadWriteAndAutoScaleWithFailoverTest-main-4] DEBUG i.p.t.system.AbstractReadWriteTest - Reading event caa1faf1-5e77-4b71-a857-baa882ab9e2e:10
Segment rebalance operation.
//reader 2 reads event 11 to 80
2019-01-21 09:20:31,785 365482 [ReadWriteAndAutoScaleWithFailoverTest-main-5] DEBUG i.p.t.system.AbstractReadWriteTest - Reading event caa1faf1-5e77-4b71-a857-baa882ab9e2e:11
2019-01-21 09:20:36,912 370609 [ReadWriteAndAutoScaleWithFailoverTest-main-5] DEBUG i.p.t.system.AbstractReadWriteTest - Reading event caa1faf1-5e77-4b71-a857-baa882ab9e2e:80

Segment rebalance happens  again...
// reader 1 reads event 81.
 2019-01-21 09:20:39,411 373108 [ReadWriteAndAutoScaleWithFailoverTest-main-4] DEBUG i.p.t.system.AbstractReadWriteTest - Reading event caa1faf1-5e77-4b71-a857-baa882ab9e2e:81

// Now the test throws 
ERROR i.p.t.system.AbstractReadWriteTest - Test exception in reading events: 
java.lang.AssertionError: Event order violated at 10 by 81

```
**Problem location**
AbstractReadWriteTest  

**Suggestions for an improvement**
Fix the issue.
[readwriteandautoscalewithfailovertest-pmiuo.zip](https://github.com/pravega/pravega/files/2785379/readwriteandautoscalewithfailovertest-pmiuo.zip)

--- a/test/system/src/test/java/io/pravega/test/system/AbstractReadWriteTest.java
+++ b/test/system/src/test/java/io/pravega/test/system/AbstractReadWriteTest.java
@@ -29,11 +29,11 @@ import io.pravega.common.concurrent.Futures;
 import io.pravega.common.util.Retry;
 import java.io.Serializable;
 import java.util.ArrayList;
-import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.UUID;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.ScheduledExecutorService;
 import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.concurrent.atomic.AtomicLong;
@@ -97,6 +97,9 @@ abstract class AbstractReadWriteTest extends AbstractSystemTest {
 
         final AtomicLong writtenEvents = new AtomicLong();
         final AtomicLong readEvents = new AtomicLong();
+        // Due to segment rebalances across readers, any reader may read arbitrary sub-sequences of events for various
+        // routing keys. This calls for a global state across readers to check the correctness of event sequences.
+        final Map<String, Long> routingKeySeqNumber = new ConcurrentHashMap<>();
 
         TestState(boolean txnWrite) {
             this.txnWrite = txnWrite;
@@ -264,7 +267,6 @@ abstract class AbstractReadWriteTest extends AbstractSystemTest {
         return CompletableFuture.runAsync(() -> {
             log.info("Exit flag status: {}, Read count: {}, Write count: {}", testState.stopReadFlag.get(),
                     testState.getEventReadCount(), testState.getEventWrittenCount());
-            final Map<String, Long> routingKeySeqNumber = new HashMap<>();
             while (!(stopFlag.get() && testState.getEventReadCount() == testState.getEventWrittenCount())) {
                 log.info("Entering read loop");
                 // Exit only if exitFlag is true  and read Count equals write count.
@@ -277,7 +279,7 @@ abstract class AbstractReadWriteTest extends AbstractSystemTest {
                         // produced them and that there are no duplicate or missing events.
                         final String[] keyAndSeqNum = event.split(RK_VALUE_SEPARATOR);
                         final long seqNumber = Long.valueOf(keyAndSeqNum[1]);
-                        routingKeySeqNumber.compute(keyAndSeqNum[0], (rk, currentSeqNum) -> {
+                        testState.routingKeySeqNumber.compute(keyAndSeqNum[0], (rk, currentSeqNum) -> {
                             if (currentSeqNum != null && currentSeqNum + 1 != seqNumber) {
                                 throw new AssertionError("Event order violated at " + currentSeqNum + " by " + seqNumber);
                             }

